cpp/src/cider/exec/nextgen/operators/RowToColumnNode.cpp

```diff
@@ -127,6 +132,171 @@ class ColumnWriter {
+                    .ret_type = JITTypeTag::VOID,
+                    .params_vector = {
+                        null_buffer.get(),
+                        values.getElemNull().get(),
+                        len->castJITValuePrimitiveType(JITTypeTag::INT32).get()}});
+          }
+          return null_buffer;
+        });
+
+    // values buffer in child
+    // get latest pointer to data buffer, will allocate data buffer on first call,
+    // and will reallocate buffer if more capacity is needed
+    auto values_bytes_len = offset_pointer[index_] *
+                            getBytesFromSQLType(expr_->get_type_info().get_subtype());
+    auto raw_data_buffer = context_.getJITFunction()->emitRuntimeFunctionCall(
+        "get_data_buffer_with_realloc_on_demand",
+        JITFunctionEmitDescriptor{
+            .ret_type = JITTypeTag::POINTER,
+            .ret_sub_type = JITTypeTag::INT8,
+            .params_vector = {
+                child_array.get(),
+                values_bytes_len->castJITValuePrimitiveType(JITTypeTag::INT32).get(),
+                context_.getJITFunction()->createLiteral(JITTypeTag::INT32, 1).get()}});
+    auto actual_raw_data_buffer = JITValuePointer(nullptr);
+    switch (expr_->get_type_info().get_subtype()) {
+      case kTINYINT:
+        actual_raw_data_buffer.replace(
+            values.getValue()->castPointerSubType(JITTypeTag::INT8));
+        break;
+      case kSMALLINT:
+        actual_raw_data_buffer.replace(
+            values.getValue()->castPointerSubType(JITTypeTag::INT16));
+        break;
+      case kINT:
+        actual_raw_data_buffer.replace(
+            values.getValue()->castPointerSubType(JITTypeTag::INT32));
+        break;
+      case kBIGINT:
+        actual_raw_data_buffer.replace(
+            values.getValue()->castPointerSubType(JITTypeTag::INT64));
+        break;
+      case kFLOAT:
+        actual_raw_data_buffer.replace(
+            values.getValue()->castPointerSubType(JITTypeTag::FLOAT));
+        break;
+      case kDOUBLE:
+        actual_raw_data_buffer.replace(
+            values.getValue()->castPointerSubType(JITTypeTag::DOUBLE));
+        break;
+      default:
+        CIDER_THROW(CiderException, std::string("Unsupported list data type"));
+        break;
+    }
+    auto cur_pointer = actual_raw_data_buffer + offset_pointer[index_];
+
+    auto data_bytes_len = len * getBytesFromSQLType(expr_->get_type_info().get_subtype());
+    context_.getJITFunction()->emitRuntimeFunctionCall(
+        "do_memcpy",
+        JITFunctionEmitDescriptor{
+            .ret_type = JITTypeTag::VOID,
+            .params_vector = {
+                raw_data_buffer->castPointerSubType(JITTypeTag::INT8).get(),
+                cur_pointer->castPointerSubType(JITTypeTag::INT8).get(),
+                data_bytes_len->castJITValuePrimitiveType(JITTypeTag::INT32).get()}});
+
+    // set arrow array length only once
+    auto tmp = context_.getJITFunction()->createLocalJITValue([this,
```

Comment:
how to do this only once without using `createLocalJITValue`, which must has a return value.